File: library/network/src/main/java/com/example/network/callback/IResponseCallback.java
package com.example.network.callback;

import androidx.annotation.NonNull;

import com.example.network.bean.ErrorBean;
import com.example.network.tag.ReqTag;

/**
 * @author fqxyi
 * @date 2018/2/27
 * 网络请求回调接口
 */
public interface IResponseCallback<T> {

    void onSuccess(@NonNull ReqTag reqTag, @NonNull T response);

    void onError(@NonNull ReqTag reqTag, @NonNull ErrorBean errorBean);

}

File: library/network/src/main/java/com/example/network/callback/IResponseCallback.kt
package com.example.network.callback

import com.example.network.bean.ErrorBean
import com.example.network.tag.ReqTag

/**
 * @author fqxyi
 * @date 2018/2/27
 * 网络请求回调接口
 */
interface IResponseCallback<T> {
    fun onSuccess(reqTag: ReqTag, response: T)
    fun onError(reqTag: ReqTag, errorBean: ErrorBean)
}